client/lib/cart-values/index.js

```diff
@@ -123,7 +124,7 @@ function isCreditCardPaymentsEnabled( cart ) {
 }
 
 function isPayPalExpressEnabled( cart ) {
-	return cart.allowed_payment_methods.indexOf( 'WPCOM_Billing_PayPal_Express' ) >= 0;
+	return cart.allowed_payment_methods.indexOf( 'WPCOM_Billing_PayPal_Express' ) >= 0 && ! config.isEnabled( 'desktop' );
```

Comment:
I wonder if the feature flag shouldn't be the first one in the condition so we don't have to read a long expression to find out that it's behind a feature gate.